paasta_tools/cli/cmds/spark_run.py

```diff
@@ -502,22 +504,22 @@ def get_docker_image(args, instance_config):
     if args.build:
         return build_and_push_docker_image(args)
     if args.image:
-        return args.image
-
-    try:
-        docker_url = instance_config.get_docker_url()
-    except NoDockerImageError:
-        print(
-            PaastaColors.red(
-                "Error: No sha has been marked for deployment for the %s deploy group.\n"
-                "Please ensure this service has either run through a jenkins pipeline "
-                "or paasta mark-for-deployment has been run for %s\n"
-                % (instance_config.get_deploy_group(), args.service)
-            ),
-            sep="",
-            file=sys.stderr,
-        )
-        return None
+        docker_url = args.image
+    else:
+        try:
+            docker_url = instance_config.get_docker_url()
+        except NoDockerImageError:
+            print(
+                PaastaColors.red(
+                    "Error: No sha has been marked for deployment for the %s deploy group.\n"
+                    "Please ensure this service has either run through a jenkins pipeline "
+                    "or paasta mark-for-deployment has been run for %s\n"
+                    % (instance_config.get_deploy_group(), args.service)
+                ),
+                sep="",
+                file=sys.stderr,
+            )
+            return None
     print(
         "Please wait while the image (%s) is pulled (times out after 5m)..."
```

Comment:
i think we still want to validate that the docker image we're going to use is actually valid (well, accessible): someone could try to point spark-run to the configs for a dev instance on a prod box and this would return a docker url that exists, but is not actually reachable due to firewall rules

Reply:
@nemacysts I didn't understand this comment. The only change which I have made here is to check `sudo docker pull` even when the args.image is provided.

Reply:
err, i think i might have just misread the code initially - this is probably fine, although we could probably speed this up by actually checking the registry api (but i guess we can always do this in the future since this should really only be affecting folks doing adhoc runs)